ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/PauseButton.tsx

```diff
@@ -53,61 +57,36 @@ type PauseFunctionModalProps = {
 function PauseFunctionModal({
   functionID,
   functionName,
-  currentVersion,
-  previousVersion,
   isPaused,
   isOpen,
   onClose,
 }: PauseFunctionModalProps) {
-  const [, pauseFunctionMutation] = useMutation(PauseFunctionDocument);
+  const [, pauseFunction] = useMutation(PauseFunctionDocument);
+  const [, unpauseFunction] = useMutation(UnpauseFunctionDocument);
   const router = useRouter();
 
   function handlePause() {
-    if (functionID && currentVersion) {
-      pauseFunctionMutation({
-        input: {
-          description: null,
-          promote: null,
-          workflowID: functionID,
-          disable: new Date().toISOString(),
-          version: currentVersion,
-        },
-      }).then((result) => {
-        if (result.error) {
-          toast.error(`${functionName} could not be paused: ${result.error.message}`);
-        } else {
-          toast.success(
-            `${result.data?.editWorkflow?.workflow.name || functionName} was successfully paused`
-          );
-          router.refresh();
-        }
-      });
-      onClose();
-    }
+    pauseFunction({ fnID: functionID }).then((result) => {
+      if (result.error) {
+        toast.error(`${functionName} could not be paused: ${result.error.message}`);
+      } else {
+        toast.success(`${functionName} was successfully paused`);
+        router.refresh();
+      }
+    });
+    onClose();
```

Comment:
question: What does the user see if the modal closes and the router hasn't been refreshed yet? Does the button still say `Pause` until the refresh and it chances to `Unpause`? I'm just wondering about the UX and any awkwardness that might exist while the request is in flight.

Reply:
The `router.refresh()` isn't actually needed -- I'll delete it. It isn't needed because the mutation returns the `Workflow` type. This causes the urql cache to invalidate and it'll refetch the data on the page.

The cache invalidation is _too_ aggressive since we use a query cache -- it'll refetch any query with `__typename__ === "Workflow"` which is the whole page. Ideally we'd switch to a normalized cache, but that's a whole project on its own